File: src/p2.rs
pub fn solve() -> i32 {
    let mut sum: i32 = 0;

    let mut i: i32 = 1;
    let mut j: i32 = 1;

    while j < 4000000 {
        sum += if j % 2 == 0 { j } else { 0 };
        let tmp: i32 = j;
        j = i + j;
        i = tmp;
    }

    return sum;
}

File: java/src/p2.java
public class p2 {
  public static void main(String[] args) {
    System.out.println("Solution of problem 2: " + solve());
  }

  private static int solve() {
    int sum = 0;
    int value = 1;
    int nextValue = 2;
    while (value < 4000000) {
      if (value % 2 == 0) {
        sum += value;
      }
      int tmp = nextValue;
      nextValue += value;
      value = tmp;
    }
    return sum;
  }
}
